### DISCUSSION TITLE
How to extract plaintext transcript from ConPTY

### DISCUSSION THREAD

**Discussion by wfjsw** [microsoft/terminal#16504] [Q&A]
I am looking for a way to dump a plain text log from a ConPTY session. Previously I would hook on the STDOUT of the attached process and do a tee-style redirection, but doing the same on a ConPTY output pipe would yield many duplicated lines when the output exceeds the vertical terminal size.

**Comment by j4james** [comment]
Right click on the window tab and select _Export Text_. You can also add your own key binding for the _Export Text_ operation if it's something you use a lot.

But if you were hoping to have the session log everything automatically as it's output, that's not supported yet. There's an open issue for that though: #14018 (also #9700).

**Reply by wfjsw** [reply]
Eh... I'm not using Windows Terminal here. Instead I am programming my own terminal window with the WPF Terminal Control and ConPTY API.
Anyway I'll look into how Windows Terminal does this. Thank you for the information.

**Reply by wfjsw** [reply]
So it called https://github.com/microsoft/terminal/blob/63c3573a13cc96bfa3d655f0f1898ae250e2fe35/src/cascadia/TerminalControl/TermControl.h#L144 and dump it all at once. This indeed suggests a way though.
